Sources/MapboxMaps/Foundation/Camera/BasicCameraAnimator.swift

```diff
@@ -0,0 +1,203 @@
+            }
+
+            guard let animation = animation else {
+                fatalError("Animation cannot be nil when starting an animation")
+            }
+
+            // Set up the short lived camera view
+            delegate.addViewToViewHeirarchy(cameraView)
+
+            var cameraTransition = CameraTransition(with: delegate.camera, initialAnchor: delegate.anchorAfterPadding())
+            animation(&cameraTransition)
+
+            propertyAnimator.addAnimations { [weak cameraView] in
+                guard let cameraView = cameraView else { return }
+                cameraView.syncLayer(to: cameraTransition.toCameraOptions) // Set up the "to" values for the interpolation
+            }
+
+            cameraView.syncLayer(to: cameraTransition.fromCameraOptions) // Set up the "from" values for the interpoloation
+            transition = cameraTransition // Store the mutated camera transition
+        }
+
+        propertyAnimator.startAnimation()
+    }
+
+    /// Starts the animation after a delay
+    /// - Parameter delay: Delay (in seconds) after which the animation should start
+    public func startAnimation(afterDelay delay: TimeInterval) {
+        delayedAnimationTimer = Timer.scheduledTimer(withTimeInterval: delay, repeats: false, block: { [weak self] (timer) in
+            guard let self = self else { return }
+            self.startAnimation()
+            timer.invalidate()
+        })
+    }
+
+    /// Pauses the animation.
+    public func pauseAnimation() {
+        propertyAnimator.pauseAnimation()
+    }
+
+    /// Stops the animation.
+    public func stopAnimation() {
+        propertyAnimator.stopAnimation(false)
+        propertyAnimator.finishAnimation(at: .current)
+    }
+
+    /// Add animations block to the animator.
+    internal func addAnimations(_ animations: @escaping (inout CameraTransition) -> Void) {
+        animation = animations
+    }
+
+    /// Add a completion block to the animator. 
+    public func addCompletion(_ completion: @escaping AnimationCompletion) {
+        let wrappedCompletion = wrapCompletion(completion)
+        propertyAnimator.addCompletion(wrappedCompletion)
+    }
+
+    internal func wrapCompletion(_ completion: @escaping AnimationCompletion) -> (UIViewAnimatingPosition) -> Void {
+        return { [weak self] animationPosition in
+            guard let self = self, let delegate = self.delegate else { return }
+            self.transition = nil // Clear out the transition being animated by this animator,
+                                  // since the animation is complete if we are here.
+            delegate.schedulePendingCompletion(forAnimator: self, completion: completion, animatingPosition: animationPosition)
+
+            // Invalidate the delayed animation timer if it exists
+            self.delayedAnimationTimer?.invalidate()
+            self.delayedAnimationTimer = nil
```

Comment:
I don't think this is the best place to invalidate the timer. Let's discuss when we talk

Reply:
This has been handled.